src/modules/mavlink/streams/OPEN_DRONE_ID_OPERATOR_ID.hpp

```diff
@@ -0,0 +1,91 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef OPEN_DRONE_ID_OPERATOR_ID_HPP
+#define OPEN_DRONE_ID_OPERATOR_ID_HPP
+
+#include <uORB/topics/vehicle_status.h>
+
+class MavlinkStreamOpenDroneIdOperatorId : public MavlinkStream
+{
+public:
+	static MavlinkStream *new_instance(Mavlink *mavlink)
+	{
+		return new MavlinkStreamOpenDroneIdOperatorId(mavlink);
+	}
+
+	static constexpr const char *get_name_static()
+	{
+		return "OPEN_DRONE_ID_OPERATOR_ID";
+	}
+	static constexpr uint16_t get_id_static()
+	{
+		return MAVLINK_MSG_ID_OPEN_DRONE_ID_OPERATOR_ID;
+	}
+
+	const char *get_name() const override { return get_name_static(); }
+	uint16_t get_id() override { return get_id_static(); }
+
+	unsigned get_size() override
+	{
+		return _vehicle_status_sub.advertised() ? MAVLINK_MSG_ID_OPEN_DRONE_ID_OPERATOR_ID_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES :
+		       0;
+
+	}
+
+private:
+	explicit MavlinkStreamOpenDroneIdOperatorId(Mavlink *mavlink)
+		: MavlinkStream(mavlink) {}
+
+	uORB::Subscription _vehicle_status_sub{ORB_ID(vehicle_status)};
+
+	bool send() override
+	{
+		mavlink_open_drone_id_operator_id_t msg{};
+		msg.target_component = 0; // 0 for broadcast
+		msg.target_system = 0;    // 0 for broadcast
+		// msg.id_or_mac // Only used for drone ID data received from other UAs.
+		msg.operator_id_type = 0;
+
+		for (uint8_t i = 0; i < 20; ++i) {
+			msg.operator_id[i] = '\0';
+		}
```

Comment:
@dagar This should be filled in with the data received from GCS if available. Same goes for `SELF_ID` and `SYSTEM`.

Since we don't have a single module like Ardupilot has that caches data received: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_OpenDroneID/AP_OpenDroneID.cpp, what would be the best way to cache these data and send them back out?

For me uORB seems to be the only solution (to connect dataflow between `mavlink_receiver` and `mavlink/streams`), but I would be interested to hear other suggestions!

Reply:
Per the mavlink definition this should be null filled if empty: https://mavlink.io/en/messages/common.html#OPEN_DRONE_ID_BASIC_ID